File: internal/tools/registry.go
package tools

import (
	"context"
	"github.com/EGjaedong/go-tiny-claw/internal/schema"
)

// Registry 定义了工具的注册与分发执行接口
type Registry interface {
	// GetAvaiablableTools 返回当前系统挂在的所有可用工具的 schema
	GetAvailableTools() []schema.ToolDefinition

	// Execute 实际执行模型请求的工具，并返回执行结果
	Execute(ctx context.Context, call schema.ToolCall) schema.ToolResult
}

File: java/src/main/java/com/egjaedong/tinyclaw/tools/Registry.java
package com.egjaedong.tinyclaw.tools;

import java.util.List;

import com.egjaedong.tinyclaw.schema.ToolCall;
import com.egjaedong.tinyclaw.schema.ToolDefinition;
import com.egjaedong.tinyclaw.schema.ToolResult;

/**
 * 工具注册与分发。对照 {@code go/internal/tools/registry.go}。
 */
public interface Registry {

    List<ToolDefinition> getAvailableTools();

    ToolResult execute(ToolCall toolCall);
}
